src/Writer/XlsxWriter.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Exporter\Writer;
+
+use PhpOffice\PhpSpreadsheet\Cell\DataType;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use PhpOffice\PhpSpreadsheet\Shared\Date;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Style\NumberFormat;
+use PhpOffice\PhpSpreadsheet\Worksheet\Worksheet;
+use PhpOffice\PhpSpreadsheet\Writer\Exception as WriterException;
+
+/**
+ * @author Willem Verspyck <willemverspyck@users.noreply.github.com>
+ */
+final class XlsxWriter implements TypedWriterInterface
+{
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var bool
+     */
+    private $showHeaders;
+
+    /**
+     * @var bool
+     */
+    private $showFilters;
+
+    /**
+     * @var Spreadsheet
+     */
+    private $spreadsheet;
+
+    /**
+     * @var Worksheet
+     */
+    private $worksheet;
+
+    /**
+     * @var int
+     */
+    private $position;
+
+    /**
+     * @throws \LogicException
+     * @throws \RuntimeException
+     */
+    public function __construct(string $filename, bool $showHeaders = true, bool $showFilters = true)
+    {
+        $this->filename = $filename;
+        $this->showHeaders = $showHeaders;
+        $this->showFilters = $showFilters;
+        $this->position = 1;
+
+        if (!class_exists(Spreadsheet::class)) {
```

Comment:
For consistency, I think we should make all the checks previous to any assignment.